code/modules/power/supermatter/supermatter.dm

```diff
@@ -1104,6 +1104,13 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 		for(var/j in 1 to zap_count)
 			supermatter_zap(target, new_range, zap_str, targets_hit, zap_flags)
 
+/obj/machinery/power/supermatter_crystal/proc/destabilize(portal_numbers)
+	var/turf/T = get_turf(src)
+	if(T)
+		qdel(src)
+		new/obj/machinery/destabilized_crystal(T)
+		explosion(T,0,round(portal_numbers/5),round(portal_numbers),1,1,1)
```

Comment:
Code after `qdel(src)` may not be executed, only `qdel` at the end of it.
Also, under which circumstances could you call this proc with the SM being in nullspace?